react-catalog/my-app/src/Product.tsx

```diff
@@ -0,0 +1,54 @@
+import { useParams, Link, useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { toDollars, readProduct, Product } from './lib';
+
+export function ProductPage() {
+  const itemId = useParams().itemID;
+  const [product, setProduct] = useState<Product>();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function work() {
+      try {
+        const prodResp = await readProduct(+itemId!);
+        setProduct(prodResp);
+      } catch {
+        return <h1>Error loading Product</h1>;
+      }
+    }
+    work();
+  });
```

Comment:
You forgot the dependency array. That's potentially a problem since it could mean making more network calls than you need. (There's no network call in this app but in a real one, `readProduct` would be a network call.)